koku/api/tags/ocp_aws/queries.py

```diff
@@ -0,0 +1,39 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+"""OCP-on-AWS Tag Query Handling."""
+from api.tags.aws.queries import AWSTagQueryHandler
+from api.tags.ocp.queries import OCPTagQueryHandler
+from reporting.models import OCPAWSCostLineItemDailySummary
+
+
+class OCPAWSTagQueryHandler(AWSTagQueryHandler, OCPTagQueryHandler):
```

Comment:
Same thing here just need to subclass the 1 tag handler.

```suggestion
class OCPAWSTagQueryHandler(AWSTagQueryHandler):
```

Reply:
Ultimately, the code calls all the way down to TagHandler for most things. So, today, inheriting from one of two classes that are subclasses of TagHandler is technically sufficient. You're correct.

But, if there are any future changes to either class, the OCP+AWS class will need that behavior, too. So, if we remove the inheritance we'll either A) need to remember the inheritance chain to know when there's a feature missing (likely to result in bugs) or B) need to add the multiple inheritance back in at that point in time.

I don't see that we gain anything other than an increased risk of bugs.

Reply:
Here too, I'm okay with multiple inheritance.